app/src/main/java/com/google/maps/android/ktx/demo/components/ScaleBar.kt

```diff
@@ -0,0 +1,289 @@
+) {
+    Text(
+        text = text,
+        fontSize = 12.sp,
+        color = textColor,
+        textAlign = TextAlign.End,
+        modifier = modifier,
+        style = MaterialTheme.typography.h4.copy(
+            shadow = Shadow(
+                color = shadowColor,
+                offset = Offset(2f, 2f),
+                blurRadius = 1f
+            )
+        )
+    )
+}
+
+/**
+ * An animated scale bar that appears when the scale of the map changes, and then disappears
+ * after [visibilityTimeoutMs]. This composable wraps [ScaleBar] with visibility animations.
+ *
+ * To use this function, pass in a [GoogleMap], which the [DisappearingScaleBar] will use for the
+ * projection of current map view to meters by registering to [GoogleMap.cameraMoveEvents]. Note
+ * that using this will override an existing listener (if any) to
+ * [GoogleMap.setOnCameraMoveListener].
+ *
+ * To avoid overriding existing listeners of [GoogleMap.setOnCameraMoveListener], implement your own
+ * observer on camera move events (e.g., val projection: Projection by googleMap.cameraMoveEvents()
+ * .collectAsState(googleMap.projection)) and animations and pass in the [Projection] to the
+ * relevant [ScaleBar] constructor.
+ */
+@ExperimentalCoroutinesApi
+@Composable
+fun DisappearingScaleBar(
+    modifier: Modifier = Modifier,
+    googleMap: GoogleMap,
+    textColor: Color = DarkGray,
+    lineColor: Color = DarkGray,
+    shadowColor: Color = Color.White,
+    visibilityTimeoutMs: Long = 3000
+) {
+    val visible = remember {
+        MutableTransitionState(true)
+    }
+
+    val projection: Projection by googleMap.cameraMoveEvents()
+        .collectAsState(googleMap.projection)
+
+    LaunchedEffect(key1 = projection) {
+        if (visible.isIdle && !visible.currentState) {
+            Log.d("scale-bar", "show")
+            // FIXME - Why doesn't the ScaleBar re-appear on the map when this executes?
+            visible.targetState = true
+        } else if (visible.isIdle && visible.currentState) {
+            delay(visibilityTimeoutMs)
+            Log.d("scale-bar", "hide")
+            visible.targetState = false
+        }
+    }
+
+    AnimatedVisibility(visibleState = visible) {
+        ScaleBar(
+            modifier = modifier,
+            projection = projection,
+            textColor = textColor,
+            lineColor = lineColor,
+            shadowColor = shadowColor
+        )
+    }
+}
+
+/**
+ * Converts the provide value in meters to the corresponding value in feet
+ * @param meters value in meters to convert to feet
+ * @return the provided meters value converted to feet
+ */
+private fun toFeet(meters: Double): Double {
+    return meters * 1000.0 / 25.4 / 12.0
+}
+
+/**
+ * Converts the provide value in feet to the corresponding value in miles
+ * @param feet value in feet to convert to miles
+ * @return the provided feet value converted to miles
+ */
+private fun toMiles(feet: Double): Double {
+    return feet / 5280
```

Comment:
Declare 5280 as const?

Reply:
Done for all constants in https://github.com/googlemaps/android-maps-ktx/pull/175/commits/77173fb34784f56305103c3fce520c96dddc3477